goshopify.go

```diff
@@ -220,6 +231,10 @@ func (c *Client) Do(req *http.Request, v interface{}) error {
 		return err
 	}
 
+	if callLimit, ok := resp.Header[callLimitHeader]; ok {
```

Comment:
I didn't know shopify actually sends the current bucket capacity in each response. I think we can utilize it to control the overall rate without explicit coordination between all the services that call shopify and contribute to API usage.